chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -125,38 +152,57 @@ def backward_gpu(self, inputs, grad_outputs):
+                    const int c = (i / n_unit % n_channel);
+                    gx = (t == -1) ? 0 : (coeff[0] * (y - (c == t)));
+                ''',
+                'softmax_crossent_bwd')(
+                    y, cupy.expand_dims(t, 1), coeff, x.shape[1], n_unit)
+        else:
+            gx = cuda.elementwise(
+                'T y, raw T w, S t, raw T coeff, S n_channel, S n_unit',
+                'T gx',
+                '''
+                    const int c = (i / n_unit % n_channel);
+                    gx = t == -1 ? 0 : coeff[0] * (y - (c == t)) * w[t];
+                ''',
+                'softmax_crossent_bwd')(
+                    y, self.class_weight, cupy.expand_dims(t, 1), coeff,
+                    x.shape[1], n_unit)
         return gx, None
 
 
 def softmax_cross_entropy(
-        x, t, use_cudnn=True, normalize=True, cache_score=True):
+        x, t, use_cudnn=True, normalize=True, cache_score=True,
+        class_weight=None):
     """Computes cross entropy loss for pre-softmax activations.
 
     Args:
-        x (Variable): Variable holding a multidimensional array whose element
-            indicates unnormalized log probability: the first axis of the
-            variable represents the number of samples, and the second axis
+        x (~chainer.Variable): Variable holding a multidimensional array whose
+            element indicates unnormalized log probability: the first axis of
+            the variable represents the number of samples, and the second axis
             represents the number of classes. While this function computes
             a usual softmax cross entropy if the number of dimensions is equal
             to 2, it computes a cross entropy of the replicated softmax if the
             number of dimensions is greater than 2.
-        t (Variable): Variable holding an int32 vector of ground truth labels.
-            If ``t[i] == -1``, corresponding ``x[i]`` is ignored.
+        t (~chainer.Variable): Variable holding an int32 vector of ground truth
+            labels. If ``t[i] == -1``, corresponding ``x[i]`` is ignored.
         normalize (bool): If ``True``, this function normalizes the cross
             entropy loss across all instances. If ``False``, it only
             normalizes along a batch size.
         cache_score (bool): When it is ``True``, the function stores result
             of forward computation to use it on backward computation. It
             reduces computational cost though consumes more memory.
+        class_weight (~numpy.ndarray or ~cupy.ndarray): An array that contains
+            constant weights that will be multiplied with the loss values along
+            with the second dimension. The shape of this array should be
+            ``(x.shape[1],)``. If this is not ``None``, each class weight
+            ``class_weight[i]`` is actually multiplied to the corresponding
+            softmax output ``y[i]`` before calculating the actual loss value.
```

Comment:
Hm, I'd like to make sure my understanding. First, `class_weight` is once broadcasted to the same shape as `log_y` that is the output of `_log_softmax` method, and then is multiplied to `log_y`. So it's actually multiplied to the log-softmax outputs, right? And why can `y[j]` be an index here: `y[j]==i`? Because `i` seems an index number, but I intended that `y` is the output vector from log-softmax operation `_log_softmax`.